src/core/attribution/markovChain.js

```diff
@@ -7,6 +7,17 @@
  */
 export type Distribution = Float64Array;
 
+export type StationaryDistributionResult = {|
+  // The final distribution after attempting to find the stationary distribution
+  // of the markov chain
+  +distribution: Distribution,
+  // How many steps were taken in this convergence attempt
+  +nIterations: number,
+  // The maximum delta between entries in the distribution as of the final
+  // markov chain step. If no steps were taken, convergenceDelta will be NaN
```

Comment:
This doesn’t seem right. What I care is the loss for the _returned
distribution_ under the action, not the _previous_ distribution. If you
return $x$, the delta should be $\norm{A x - x}_\infty$, which is well
defined even if $x$ was initialized to the uniform distribution and
never modified. No?

Reply:
I don't understand, can you please explain more simply? What is the `A` in `A x - x`, where does the sub infinity factor in?

Reply:
Sorry, was writing in a bit of shorthand, forgot to expand.

$A$ is the Markov chain transition matrix. If $x$ is a distribution
vector, then $A x$ is the distribution after one step in the Markov
chain. (`sparseMarkovChainAction(A, x)` computes `A x`.)

$\norm{\delta}_\infty$ is the [∞-norm][un] of $\delta$; equivalently,
the largest value in $\delta$.

Basically, if $x$ is actually a stationary distribution for the Markov
chain given by $A$, then $A x = x$ by definition. So $A x - x$ is a
vector that will be “smaller” when $x$ is a better approximation of a
stationary distribution. That is, $\norm{A x - x}$ is the loss function
that we’re trying to minimize. In particular, the norm that we choose is
the max-norm, though it would also be valid to minimize another norm
like the Euclidean (sum-of-squares) norm.

The purpose of returning the `convergenceDelta` (i.e., loss) to the
client is so that the client can get an idea of how good the returned
distribution is. But as documented and implemented, you’re returning the
final distribution but the loss corresponding to the penultimate
distribution. My claim is that the returned loss should actually be the
loss of the distribution that you return, not of a different
distribution.

[un]: https://en.wikipedia.org/wiki/Uniform_norm